src/fmcore/prompt_tuner/dspy/optimizers/miprov2_optimizer.py

```diff
@@ -0,0 +1,125 @@
+from typing import Dict, Optional, List, Callable
+
+import dspy
+from dspy.teleprompt import MIPROv2
+
+from fmcore.prompt_tuner import OptimizerType
+from fmcore.prompt_tuner.dspy.datasets.base_dataset import DspyDataset
+from fmcore.prompt_tuner.dspy.lm_adapters.dspy_adapter import DSPyLLMAdapter
+from fmcore.prompt_tuner.dspy.optimizers.base_dspy_optimizer import (
+    BaseDspyOptimizer,
+)
+from fmcore.prompt_tuner.dspy.utils.dspy_utils import DSPyUtils
+from fmcore.prompt_tuner.evaluator import BaseEvaluator
+from fmcore.prompt_tuner.types.optimizer_types import BaseOptimizerConfig
+from fmcore.prompt_tuner.types.prompt_tuner_types import PromptTunerConfig
+from fmcore.utils.introspection_utils import IntrospectionUtils
+
+
+class MIPROV2Optimizer(BaseDspyOptimizer):
```

Comment:
i see no meaning to this class. its literally utility class to create teh optimizer and it has nothign to do with the optimizer itself.... A factory for creatign dspy optimizer and a factory to create request params to compile should do

Reply:
that way ur tuner will use a dspy optimizer rather than another abstration of it